platform/ios/src/MGLPinAnnotationView.m

```diff
@@ -0,0 +1,96 @@
+#import "MGLPinAnnotationView.h"
+
+@implementation MGLPinAnnotationView
+
+- (instancetype)initWithReuseIdentifier:(nullable NSString *)reuseIdentifier {
+    self = [super initWithReuseIdentifier:reuseIdentifier];
+    _shadowColor = [UIColor blackColor];
+    _pinColor = [UIColor blueColor];
+    _strokeColor = [UIColor whiteColor];
+    _innerColor = [UIColor whiteColor];
+    self.backgroundColor = [UIColor clearColor];
+
+    return self;
+}
+
+- (void)drawRect:(CGRect)rect {
+    [self drawMarkerWithFrame:rect innerColor:_innerColor shadowColor:_shadowColor pinColor:_pinColor strokeColor:_strokeColor];
+}
+
+- (void)setShadowColor:(UIColor *)shadowColor {
+    _shadowColor = shadowColor;
+    [self setNeedsDisplay];
+}
+
+- (void)setPinColor:(UIColor *)pinColor {
+    _pinColor = pinColor;
+    [self setNeedsDisplay];
+}
+
+- (void)setStrokeColor:(UIColor *)strokeColor {
+    _strokeColor = strokeColor;
+    [self setNeedsDisplay];
+}
+
+- (void)setInnerColor:(UIColor *)innerColor {
+    _innerColor = innerColor;
+    [self setNeedsDisplay];
+}
+
+- (void)drawMarkerWithFrame:(CGRect)frame innerColor:(UIColor*)innerColor shadowColor:(UIColor*)shadowColor pinColor:(UIColor*)pinColor strokeColor:(UIColor*)strokeColor
+{
+    //// General Declarations
+    CGContextRef context = UIGraphicsGetCurrentContext();
+    
+    //// Group
+    {
+        //// Oval 2 Drawing
+        CGContextSaveGState(context);
+        CGContextSetAlpha(context, 0.1);
+        
+        UIBezierPath* oval2Path = [UIBezierPath bezierPathWithOvalInRect: CGRectMake(CGRectGetMinX(frame) + 10, CGRectGetMinY(frame) + 40.61, 19, 8)];
+        [shadowColor setFill];
+        [oval2Path fill];
+        
+        CGContextRestoreGState(context);
+        
+        
+        //// Group 2
+        {
+            //// Bezier 2 Drawing
+            UIBezierPath* bezier2Path = [UIBezierPath bezierPath];
```

Comment:
The paths for “Bezier 2” and “Bezier 3” appear to be identical — can this be refactored/simplified?